src/pages/PullRequestPage/subroute/FilesChangedTab/FilesChangedTab.jsx

```diff
@@ -30,11 +31,21 @@ function hasReportWithoutChanges({
 
 function FilesChangedTab() {
   const { data, isLoading } = useImpactedFilesTable()
+  const isFirstPull =
+    data?.compareWithBaseType === ComparisonReturnType.FIRST_PULL_REQUEST
 
   if (isLoading) {
     return <Loader />
   }
 
+  if (isFirstPull) {
```

Comment:
I've seen Nick start to do some of the error handling at the hook level, for example here https://github.com/codecov/gazebo/blob/main/src/services/pulls/usePulls.tsx#L235, that I think it's a nice way to clean up the react code we see. I don't think this is necessarily the time to make this change here since you'd have some error handling at the hook level and the rest here, but more of a food for thought to keep this in mind for future work

Reply:
hmmm nice, could be good to clean up later